File: app/src/main/java/org/thoughtcrime/securesms/util/BottomSheetUtil.java
package org.thoughtcrime.securesms.util;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

public final class BottomSheetUtil {

  public static final String STANDARD_BOTTOM_SHEET_FRAGMENT_TAG = "BOTTOM";

  private BottomSheetUtil() {}

  /**
   * Show preventing a possible IllegalStateException.
   */
  public static void show(@NonNull FragmentManager manager,
                          @Nullable String tag,
                          @NonNull BottomSheetDialogFragment dialog)
  {
    FragmentTransaction transaction = manager.beginTransaction();
    transaction.add(dialog, tag);
    transaction.commitAllowingStateLoss();
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/util/BottomSheetUtil.kt
package org.thoughtcrime.securesms.util

import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.fragment.app.FragmentManager
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import org.thoughtcrime.securesms.R

object BottomSheetUtil {
  const val STANDARD_BOTTOM_SHEET_FRAGMENT_TAG = "BOTTOM"

  /**
   * Show preventing a possible IllegalStateException.
   */
  @JvmStatic
  fun show(
    manager: FragmentManager,
    tag: String?,
    dialog: BottomSheetDialogFragment
  ) {
    manager.beginTransaction().apply {
      add(dialog, tag)
      commitAllowingStateLoss()
    }
  }

  fun BottomSheetDialogFragment.requireCoordinatorLayout(): CoordinatorLayout {
    return requireDialog().findViewById(R.id.coordinator)
  }
}
